installation/gcp/dm/firewall-template.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Creates the firewall."""
+
+
+def GenerateConfig(context):
+    """Creates the firewall"""
+    
+    resources = [
+        {
+            'name': f"allow-ssh-to-vmclarity-server-{context.env['deployment']}",
+            'type': 'compute.v1.firewall',
+            'properties': {
+                'network': '$(ref.' + context.properties['network'] + '.selfLink)',
+                'direction': "INGRESS",
+                'sourceRanges': ['0.0.0.0/0'],
+                'allowed': [{
+                    'IPProtocol': 'TCP',
+                    'ports': [22]
+                }]
+            }
+        },
+        {
+            'name': f"allow-scanner-to-vmclarity-server-{context.env['deployment']}",
+            'type': 'compute.v1.firewall',
+            'properties': {
+                'network': '$(ref.' + context.properties['network'] + '.selfLink)',
+                'direction': 'INGRESS',
+                'sourceRanges': ['10.128.0.0/9'],
+                'targetTags': ["vmclarity-control-plane"],
+                'allowed': [{
+                    'IPProtocol': 'TCP',
+                    'ports': ["0-65535"]
```

Comment:
Why enabling all ports instead the specific ports we listen on (8888)?

Reply:
This specifically allows access between the scanner and the control plane, I opted for the easier choice to allow all ports, then if we add any other components like grype-server etc the installer doesn't need changing. This is a little different to the other installers but we can align them if we want.